spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -1304,6 +1313,82 @@ public void testDeleteToCustomWapBranchWithoutWhereClause() throws NoSuchTableEx
         });
   }
 
+  @Test
+  public void testDeleteFilterWithSystemFuncOnPartitionedTable() {
+    Assumptions.assumeThat(catalogName).isNotEqualTo("spark_catalog");
+
+    Map<String, String> tableProperties = Maps.newHashMap();
+    tableProperties.put(DEFAULT_FILE_FORMAT, fileFormat);
+    tableProperties.put(WRITE_DISTRIBUTION_MODE, distributionMode);
+    tableProperties.put(SPARK_WRITE_PARTITIONED_FANOUT_ENABLED, String.valueOf(fanoutEnabled));
+    tableProperties.putAll(extraTableProperties());
+    createPartitionedTable(spark, tableName, "years(ts)", tableProperties);
+
+    String date = "2018-12-21";
+    String remainSql =
+        sqlFormat(
+            "SELECT * FROM %s WHERE %s.system.years(ts) != %s.system.years(date('%s')) ORDER BY id",
```

Comment:
This should not rely on constant folding to produce the expected rows. If it does, then problems could affect both. Instead, this should calculate the value directly and embed it in the SQL.

Reply:
Here, I want to test the constant system function could fold as well.

Reply:
It's fine to have a test that the function is foldable, but I don't think it should be used to produce expected rows. Instead, use a constant here and test that the folding is done elsewhere.

Reply:
OK, updated.